### ISSUE TITLE
[java] New rule: AvoidVarForShortType

### ISSUE THREAD

**Issue by AB-xdev** [pmd/pmd#6903]
**Proposed Rule Name:** AvoidVarForShortType

**Proposed Category:** Code Style

**Description:**

_Original backstory and arguments can be found in https://github.com/xdev-software/java-setup-template/issues/3_

I'd basically like a rule that flags usage of the `var` keyword when the (normally) written type would be below a certain length.
Writing `var x = 0` instead of `int x = 0` makes not a lot of sense and just makes reading/reviewing the code (without an IDE) harder.
Also most IDEs have a "variable extraction feature" that automatically adds the correct type.

Using `var` for something like `Iterator<? extends Map.Entry<? extends MySuperCoolCacheKey, ? extends MySuperCoolValue>>` however is a lot reasonable IMHO.

For further reading I can also recommend [OpenJDK's Local Variable Type Inference CSG](https://openjdk.org/projects/amber/guides/lvti-style-guide).

**Code Sample:** 

```java

// Could have just used an `int`
// Will create an `int` however you might want a `byte` or `short`
var x = 0;

// DANGEROUS: infers as List<Object>
var list = List.of();

// What is customers? Maybe a List? a Set?
var customers = dbconn.executeQuery("SELECT ...");
// A dozen lines of other code go here...
customers.map(...).toList(); // Nevermind it was a Stream...
```

**Possible Properties:**
* `lengthLimit` - The rule will be triggered when the length of the type is below this limit

**Comment by AB-xdev** [comment]
As I already did a reference implementation I will send a PR for this

**Comment by UncleOwen** [comment]
>// DANGEROUS: infers as List\<Object\>
var list = List.of();

Maybe find a better example here. The list returned by `List.of()` is empty and immutable, so this is safe.

But I like the idea, awaiting your PR!
